tools/liveimage-mount

```diff
@@ -63,43 +79,74 @@ def call(*popenargs, **kwargs):
     return rc
 
 
-def rcall(args, env=None):
+def rcall(args, stdin=None, raise_err=True, env=None):
+    """Return stdout, stderr, & returncode from a subprocess call."""
+
+    out, err, p, environ = '', '', None, None
     if env:
         environ = os.environ.copy()
         environ.update(env)
-    else:
-        environ = None
     try:
-        p = subprocess.Popen(args, stdout=subprocess.PIPE,
-                             stderr=subprocess.PIPE, env=environ)
-        out, err = p.communicate()
-    except OSError, e:
-        raise CreatorError(u"Failed to execute:\n'%s'\n'%s'" % (args, e))
-    except:
-        raise CreatorError(u"""Failed to execute:\n'%s'
-            \renviron: '%s'\nstdout: '%s'\nstderr: '%s'\nreturncode: '%s'""" %
-            (args, environ, out, err, p.returncode))
+        p = subprocess.Popen(args, stdin=subprocess.PIPE, env=environ,
+                             stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = p.communicate(stdin)
+    except OSError as e:
+        out, err = out, u'Failed executing:\n%s\nerror: %s' % (args, e)
+        if raise_err:
+            raise LiveImageMountError(u'Failed executing:\n%s\nerror: %s' %
+                                      (args, e))
+    except Exception as e:
+        out, err = (out, u'Failed to execute:\n%s\nerror: %s' %
+                          (args, e))
+        if raise_err:
+            raise LiveImageMountError(u'''Failed to execute:\n%s
```

Comment:
Triple quotes have the advantage of including (the usually hidden) newlines and indentations from the editor in the code parsing.

This advances Python's style goals of making more readable code by minimizing the _visible_ parsing & formatting code symbols that can distract the eye from the desired visible content.

See, for example,
```python
#!/usr/bin/python -tt

def main():

    a = '!!!'

    # indentations provided by \t, parentheses needed for line grouping.
    s = ('\n\tfish'
         '\n\tfry'
         '\n\t')

    print s + a

    # indentations from editor included by line continuations.
    s = '\n\
        fish\n\
        fry\n\
        '

    print s + a

    # indentations and newlines from editor included by triple quotes.
    s = '''
        fish
        fry
        '''

    print s + a

if __name__ == '__main__':
    main()
```
This is especially helpful when working near a specific line length limit.

Here, the 18 pixels (on my monitor) for the pair of ```'''``` (triple single quotes) is far less visual 'noise' than all the backslashes, n's, t's, & parentheses placed right next to the desired visible content.